crates/eth-rpc/src/rpc.rs

```diff
@@ -0,0 +1,42 @@
+use jsonrpsee::RpcModule;
+use kakarot_rpc_core::client::config::{JsonRpcClientBuilder, StarknetConfig};
+use kakarot_rpc_core::client::KakarotClient;
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::JsonRpcClient;
+
+use crate::api::eth_api::EthApiServer;
+use crate::api::kakarot_custom_api::KakarotCustomApiServer;
+use crate::servers::eth_rpc;
+use crate::servers::eth_rpc::KakarotEthRpc;
+use crate::servers::kakarot_custom_rpc::{self, KakarotCustomRpc};
+
+pub struct KakarotRpc {
+    eth_rpc_module: RpcModule<KakarotEthRpc<JsonRpcClient<HttpTransport>>>,
+    kakarot_custom_rpc_module: RpcModule<KakarotCustomRpc<JsonRpcClient<HttpTransport>>>,
+}
+
+impl KakarotRpc {
+    pub fn new(starknet_config: StarknetConfig) -> Self {
```

Comment:
a problem I see here is that we're using 2 different `KakarotClient` instances, one for each API, but they're the same client! 
It'll end up being quite inefficient

Possible workarounds:
Having Arc clones of KakarotClient passed around?

You'll have to investigate if that doesn't work out -> our main philosophical challenge here is: each api will probably point to core crate, thus kakarot client, how can we not duplicate kakarot client instances for each separate api

Reply:
Resolved with the new code changes I pushed.

<img width="788" alt="image" src="https://github.com/kkrt-labs/kakarot-rpc/assets/41180869/6dddbba7-29fd-41f8-a1a2-7d21fedb5631">